src/StoragelessSession/Session/Data.php

```diff
@@ -25,6 +25,28 @@ private function __construct(array $data, array $metadata)
         $this->metadata = $metadata;
     }
 
+    public static function fromDecodedTokenData(\stdClass $data)
+    {
+        return self::fromTokenData(self::convertStdClassToUsableStuff($data), []);
+    }
+
+    private static function convertStdClassToUsableStuff(\stdClass $shit)
```

Comment:
We could avoid this boring stuff by passing `true` as second argument to `json_decode()` [there](https://github.com/lcobucci/jose-parsing/blob/1ea9a54dcbf3a92240313b3dc1dde6e816576264/src/Decoder.php#L28).